### PR TITLE
Clarify slash commands implementation

### PR DESCRIPTION
**Pull request by Zaitara**
This PR clarifies the slash commands and interaction implementation by improving the documentation and renames confusing `create(_global)_application_commands(_permissions)` methods to `set(_global)_application_commands(_permissions)`.

This has been tested.

### CODE REVIEW

File: src/builder/create_application_command_permission.rs [hunk 1]
```diff
@@ -71,7 +73,7 @@ impl CreateApplicationCommandPermissions {
         self
     }
 
-    /// Creates a new permission for the interaction.
+    /// Creates permissions(s) for the application command.
```
Comment (Flat) [inline src/builder/create_application_command_permission.rs]:
```suggestion
    /// Creates permissions for the application command.
```
I've never seen someone use singular permission when setting permissions in computer terminology even if it is a singular permission.

File: src/builder/create_application_command_permission.rs [hunk 2]
```diff
@@ -86,7 +88,7 @@ impl CreateApplicationCommandPermissions {
         self
     }
 
-    /// Adds a permission for the interaction.
+    /// Adds permission(s) for the application command.
```
Comment (Flat) [inline src/builder/create_application_command_permission.rs]:
```suggestion
    /// Adds permissions for the application command.
```

File: src/builder/create_application_command_permission.rs [hunk 3]
```diff
@@ -101,7 +103,7 @@ impl CreateApplicationCommandPermissions {
         self
     }
 
-    /// Sets all the permissions for the interaction.
+    /// Sets permission(s) for the application command.
```
Comment (Flat) [inline src/builder/create_application_command_permission.rs]:
```suggestion
    /// Sets permissions for the application command.
```

File: src/builder/create_application_command_permission.rs [hunk 4]
```diff
@@ -153,7 +157,7 @@ impl CreateApplicationCommandPermissionsData {
         self
     }
 
-    /// Sets all the permissions for the interaction.
+    /// Sets a permission for the application command.
```
Comment (Flat) [inline src/builder/create_application_command_permission.rs]:
```suggestion
    /// Sets permissions for the application command.
```

**Review by Flat — COMMENTED** [review]
It may also be nice to add an example of handling an `Interaction` in the handler such as the ping command, just replying "pong".

File: src/builder/create_application_command_permission.rs (hunk 1 above)
Comment (Zaitara) [reply src/builder/create_application_command_permission.rs]: I wanted to make a distinction, even if it is not a common terminology. I'll change that

File: examples/e14_slash_commands/src/main.rs [hunk 5]
```diff
@@ -12,21 +22,97 @@ struct Handler;
...
+                                    "I hope that you brought a controller to play together!",
+                                    "game",
+                                )
+                        })
+                })
+        })
+        .await;
 
-        println!("I have the following global slash command(s): {:?}", interactions);
+        println!("I have now the following global slash commands: {:#?}", commands);
```
Comment (sbrocket) [inline examples/e14_slash_commands/src/main.rs]: nit: "I now have ..."

**Comment by Zaitara** [comment]
This PR will needs #1369 to get merged as the example will be outdated.

**Comment by Oyabi** [comment]
You can maybe indicate that it can take 1 hour for global commands to be available, but it's instant for guild commands. [Source here](https://canary.discord.com/developers/docs/interactions/slash-commands#registering-a-command) search for the first and second yellow frame or ctrl+f "hour".

Maybe implement a guild command for ping and another one for global command?

**Comment by Zaitara** [comment]
> You can maybe indicate that it can 1 hour for global commands to be available, but it's instant for guild commands. [Source here](https://canary.discord.com/developers/docs/interactions/slash-commands#registering-a-command) search for the first and second yellow frame or ctrl+f "hour".

Well it is already specified on the `ApplicationCommand::create_global_application_command` method, see [here](https://serenity-rs.github.io/serenity/current/serenity/model/interactions/struct.ApplicationCommand.html#method.create_global_application_command).

> Maybe implement a guild command for ping and another one for global command?

Indeed in the example it is only shown for a global command. I'll add the guild command creation in the example

**Review by arqunis — APPROVED** [review]

**Pull request merged**